### PR TITLE
LeveledLogger: Add doc and prioritize it over Logger

### PR DESCRIPTION
**Pull request by ppai-plivo**
* Document LeveledLogger better to make it clear to developers that the
  second variadic argument accepts key-value pairs like hci-log or zap
  logger and not Printf style logging.
* When the passed logger implements both Logger and LeveledLogger
  interfaces, prioritize LeveledLogger over Logger. Switch statements
  evaluate cases from top to bottom.

### CODE REVIEW

**Comment by ppai-plivo** [comment]
cc @jefferai for some 👀   on this. Thank you!

File: client.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/hashicorp/go-cleanhttp"
+	cleanhttp "github.com/hashicorp/go-cleanhttp"
```
Comment (jefferai) [inline client.go:42]: This is an unnecessary change.

File: client.go [hunk 2]
```diff
@@ -501,10 +505,10 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 
 	if logger != nil {
 		switch v := logger.(type) {
-		case Logger:
```
Comment (jefferai) [inline client.go:504]: I'm not sure that this was an accident. It might have been prioritized to keep behavior the same for backwards compat. @evanphx would know more.

Comment (evanphx) [reply client.go:504]: Looks like he just wanted to prioritize the LevelLogger over the generic one. That seems like a good change since LevelLogger is more specific.

File: client.go (hunk 1 above)
Comment (ppai-plivo) [reply client.go:42]: It is unrelated but is intentional as it fixes the auto-removal of this import by goimports because in this case, the import path does not match the package name. Ref: https://github.com/golang/go/issues/28428

This slightly improves readability with no functional repercussions. I will remove this if you insist. Thank you.

cc @jefferai @evanphx

Comment (jefferai) [reply client.go:42]: So that's strange because imports prefixed by `go-` are auto-added just fine for me. That's why I suggested this wasn't necessary. I actually just tried taking a file and removing the `cleanhttp` and running `gomiports` on the file and it leaves it alone. Is it possible you're using a very old version of goimports or that you're not using go modules? IIRC this started working properly right around when go modules dropped on the scene.

File: client.go (hunk 2 above)
Comment (jefferai) [reply client.go:504]: Sounds good then.

**Comment by jefferai** [comment]
Thanks!

File: client.go (hunk 1 above)
Comment (ppai-plivo) [reply client.go:42]: Sorry, my bad. You're right. I was using go modules but I had an older version of goimports.
